### ISSUE TITLE
Ceph CSI Lifecycle Hook fails

### ISSUE THREAD

**Issue by galexrt** [rook/rook#4427]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**

The lifecycle `preStop` hook fails on container stop / exit because `/bin/sh` is not found in the driver registrar container image for some reason.

**Expected behavior:**

The lifecycle `preStop` to succeed and remove the csi driver sockets successfully.

**How to reproduce it (minimal and precise):**

**File(s) to submit**:

* Cluster CR (custom resource), typically called `cluster.yaml`, if necessary
* Operator's logs, if necessary
* Crashing pod(s) logs, if necessary

```
  Normal   Killing            7m36s                 kubelet, my-node1  Stopping container driver-registrar
  Normal   Killing            7m36s                 kubelet, my-node1  Stopping container liveness-prometheus
  Normal   Killing            7m36s                 kubelet, my-node1  Stopping container csi-rbdplugin
  Warning  FailedPreStopHook  7m35s                 kubelet, my-node1  Exec lifecycle hook ([/bin/sh -c rm -rf /registration/csi-rbdplugin /registration/csi-rbdplugin-reg.sock]) for Container "driver-registrar" in Pod "csi-rbdplugin-4j6w5_rook-ceph(82b9762f-672a-4def-b094-04ba03aeb086)" failed - error: command '/bin/sh -c rm -rf /registration/csi-rbdplugin /registration/csi-rbdplugin-reg.sock' exited with 126: , message: "OCI runtime exec failed: exec failed: container_linux.go:345: starting container process caused \"exec: \\\"/bin/sh\\\": stat /bin/sh: no such file or directory\": unknown\r\n"
  Warning  FailedKillPod      5m35s                 kubelet, my-node1  error killing pod: failed to "KillContainer" for "csi-rbdplugin" with KillContainerError: "rpc error: code = Unknown desc = operation timeout: context deadline exceeded"
  Warning  FailedSync         36s (x15 over 3m35s)  kubelet, my-node1  error determining status: rpc error: code = DeadlineExceeded desc = context deadline exceeded
```

**Environment**:
* OS (e.g. from /etc/os-release): Ubuntu 18.04.3 LTS
* Kernel (e.g. `uname -a`): 4.15.0-70 generic
* Cloud provider or hardware configuration: Bare metal on-premise
* Rook version (use `rook version` inside of a Rook Pod): v1.1.7
* Storage backend version (e.g. for ceph do `ceph -v`): v14.2.4
* Kubernetes version (use `kubectl version`): v1.15.5
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift): Kubernetes
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/master/ceph-toolbox.html)): `HEALTH_WARN too few PGs per OSD` (should not be relevant here)

**Comment by galexrt** [comment]
Ping @Madhu-1 could you please take a look at this? Seems like a minor thing but potentially "breaking" for the kubelet as it doesn't see the "driver socket" removed and is timing out "all the time".

**Comment by Madhu-1** [comment]
@galexrt i will look into it

**Comment by Madhu-1** [comment]
posted upstream PR https://github.com/kubernetes-csi/node-driver-registrar/pull/61 in sidecar to fix this issue

**Comment by Madhu-1** [comment]
The upstream patch is not merged yet

**Comment by nirav-chotai** [comment]
```
driver-registrar:
    Container ID:
    Image:         quay.io/k8scsi/csi-node-driver-registrar:v2.0.0-rc1
```

I am using the above image, still facing the same issue

```
57m         Warning   FailedPreStopHook        pod/csi-cephfsplugin-dr8cc                                        Exec lifecycle hook ([/bin/sh -c rm -rf /registration/rook-ceph.cephfs.csi.ceph.com /registration/rook-ceph.cephfs.csi.ceph.com-reg.sock]) for Container "driver-registrar" in Pod "csi-cephfsplugin-dr8cc_rook-ceph(ded38899-4a57-475c-9b8c-318579f1823c)" failed - error: command '/bin/sh -c rm -rf /registration/rook-ceph.cephfs.csi.ceph.com /registration/rook-ceph.cephfs.csi.ceph.com-reg.sock' exited with 126: , message: "OCI runtime exec failed: exec failed: container_linux.go:348: starting container process caused \"exec: \\\"/bin/sh\\\": stat /bin/sh: no such file or directory\": unknown\r\n"
```

```
$ kubectl -n rook-ceph exec -it csi-cephfsplugin-4xfvd -c driver-registrar -- /bin/sh
OCI runtime exec failed: exec failed: container_linux.go:348: starting container process caused "exec: \"/bin/sh\": stat /bin/sh: no such file or directory": unknown
command terminated with exit code 126
```

**Comment by Madhu-1** [comment]
@nirav-chotai this need template change in rook. Once the above-mentioned node driver release is done will make necessary changes in rook.

**Comment by nirav-chotai** [comment]
I see it was merged and released in v2.0.0-rc1, that's why I tried that out to fix the issue.

**Comment by Madhu-1** [comment]
@nirav-chotai once #6141 is merged this can be tested again.
